include/cadmium/concept/coupled_model_assert.hpp

```diff
@@ -169,7 +169,7 @@ namespace cadmium {
 
         //asserting the submodels are proper coupled or atomic modes.
         template<typename MODELs, int S>
-        struct assert_each_model {
+        struct assert_each_pdevs_submodel {
```

Comment:
I would ensure name format consistency, in line 40 you started with pdevs_ so why in this line to put pdevs in the middle of the name?